api/src/test/java/gov/cms/ab2d/api/controller/TestUtil.java

```diff
@@ -0,0 +1,111 @@
+package gov.cms.ab2d.api.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import gov.cms.ab2d.common.model.Sponsor;
+import gov.cms.ab2d.common.model.User;
+import gov.cms.ab2d.common.repository.SponsorRepository;
+import gov.cms.ab2d.common.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+public class TestUtil {
+
+    public static final String TEST_USER = "EileenCFrierson@example.com";
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private SponsorRepository sponsorRepository;
+
+    private Map<String, String> headerMap;
+
+    // This will expire in 3600 seconds
+    public Map<String, String> setupToken() throws IOException, InterruptedException {
+        setupUser();
+
+        // Tests run 1 at a time, this may pose an issue if they ever run concurrently
+        if(headerMap != null) {
+            return headerMap;
+        }
+
+        headerMap = new HashMap<>();
+
+        var values = new HashMap<String, String>() {{
+            put("grant_type", "password");
+            put("username", TEST_USER);
+            put("password", "Te$t2019");
+            put("scope", "openid");
+        }};
+
+        HttpClient client = HttpClient.newHttpClient();
+        HttpRequest request = HttpRequest.newBuilder()
```

Comment:
The tests can't get tokens from Okta. Making tests dependent on an _external_ system makes them fragile. We have to make our own self-signed JWT here.